### ISSUE TITLE
Basic initial build

### ISSUE THREAD

**Issue by msummers42** [kojiromike/docker-gentoo-uwsgi#1]
This is one approach, though it needs a fair bit of tweaking.
```
FROM gentoo/portage:latest as portage
FROM gentoo/stage3-amd64-hardened-nomultilib:latest

COPY --from=portage /usr/portage /usr/portage

RUN USE="python" emerge -qv uwsgi && \
  emerge --info uwsgi
```

**Comment by kojiromike** [comment]
I put a Dockerfile into the `context` subdirectory, but right now it's focused on building an up-to-date system. The image seems to still be using python 3.5 and has a lot of old stuff.

**Comment by msummers42** [comment]
In that case, we will need to create a binpkg host and customize our own "stage3", which is fine and dandy. So I was looking at GCC-7.3 for this sort of thing a few weeks back and noticed a `mtune=intel` which enabled some nice features for Haswell and Broadwell(?) CPUs. I am not sure we want that, however we will lose some performance and stability if we do not enable some ARCH-specific features.

**Comment by kojiromike** [comment]
We can create a fat docker image full of binpkgs and use that to provide binpkgs to downstream build images using the `COPY --from` approach like Gentoo does with the `portage` image. What do you think?

**Comment by msummers42** [comment]
Not using `docker build`, due to the restricted capabilities (at least TRACE is forbidden, which is needed to build GCC). There are workaround using `docker run` to do a bunch of actions in various phases. I would be interested in looking at other IMG builders for this, perhaps.

**Comment by msummers42** [comment]
Of course, we could just use Opkins to build our binpkgs for various architectures which would be useful in various places.

**Comment by kojiromike** [comment]
I seem to have been able to build gcc in an image /shrug

**Comment by msummers42** [comment]
Perhaps it was glibc then, I cannot recall. Whatever it was, it got in the way massively and I was sufficiently annoyed that I stopped playing with it.
